app/jobs/importo/import_job.rb

```diff
@@ -1,23 +1,52 @@
 module Importo
   class ImportJob
     include Sidekiq::Job
+    sidekiq_options retry: 5
+    # queue_as :integration
+    queue_as Importo.config.queue_name
+
+    sidekiq_retries_exhausted do |msg, _e|
+      attributes = msg['args'][0]
+      index = msg['args'][1]
+      import_id = msg['args'][2]
+      signal_id = msg['args'][3]
+
+      execute_row(attributes, index, import_id, signal_id, true, msg['bid'])
+    end
 
     def perform(attributes, index, import_id, signal_id)
+      self.class.execute_row(attributes, index, import_id, signal_id, false, bid)
+
+      # puts "Working within batch #{bid}"
+      # batch.jobs do
+      # add more jobs
+      # end
+    end
+
+    def self.execute_row(attributes, index, import_id, signal_id, last_attempt, bid)
```

Comment:
self. ? do we require class method

Reply:
yes, its required when we need to call from sidekiq_retries_exhausted hook